custom_components/bosch_alarm/sensor.py

```diff
@@ -49,10 +49,39 @@ async def async_added_to_hass(self):
 
     async def async_will_remove_from_hass(self):
         self._panel.history_observer.detach(self.schedule_update_ha_state)
+class PanelFaultsSensor(SensorEntity):
+    def __init__(self, panel):
+        self._panel = panel
+        self._attr_device_info = device_info_from_panel(panel)
```

Comment:
I think we extract some boilerplate into a common sensor base class.
This, should_poll, the observer handling -- maybe something else.